src/main/resources/explanations/challenge24_reason.adoc

```diff
@@ -6,3 +6,5 @@ By now you know that it is not recommended to copy every primitive of an example
 Please note that copying keys from specs/examples does not only hold for HMACs, it holds for any cryptographic operation (signing, encryption, decryption, etc.).
 
 Still need to generate a key? Make sure you use a Secure Random generator and the right library for your runtime to generate the key instead.
+
+To make it even safer: make sure you use a key generation and/or derivation setup in which every new message has a different key. That way you have "Perfect Forward secrecy". This will ensure that older messages cannot be decrypted when one key is compromised.
```

Comment:
```suggestion
To make it even safer, use a key generation and/or derivation setup in which every new message has a different key. That way, you have "Perfect Forward secrecy". This will ensure that older messages cannot be decrypted when one key is compromised.
```